### ISSUE TITLE
Geocomplete - Is there a way to get 'establishment' AND 'address' types within the same field?

### ISSUE THREAD

**Issue by CharlieEtienne** [cheesegrits/filament-google-maps#87]
Hi,

In the `getTypes()` method in `Geocomplete` class, it defaults to 'geocode', making impossible to get results from Establishments AND Addresses.

```php
public function getTypes(): array
    {
        $types = $this->evaluate($this->types);

        if (count($types) === 0) {
            // $types = ['geocode'];
        }

        return $types;
    }
```

If I comment out the default value it works like I want, but could you provide a cleaner solution?

Thanks

**Comment by kengraversen** [comment]
Hi @CharlieEtienne

I couldn't figure out why I didn't get any business places and was looking for a solution.
Your suggestion looks great to me because it would default to all types.
A workaround seems to be to specify more or all "table 3" types. The Google documentation says it won't work, but it works for me.
Here is what I specify to include three types:
'''autocomplete( 'geo_location', types: [ 'geocode', 'establishment', 'address'])'''
A note so you may have a workaround until the change request is approved and merged.

**Comment by cheesegrits** [comment]
You should be able to specify the types you want to use in the ->autocomplete() method, using the 'types' argument, which takes an array of types, constrained according to the Google PIA docs:

https://developers.google.com/maps/documentation/javascript/place-autocomplete#constrain-place-types

**Comment by CharlieEtienne** [comment]
the problem is that it allows to constrain, but I don't want to constrain. I want all types of results, and that's not possible when you specify a type property with any combination (according to my tests) I read all the docs 3 times and tried all sort of combinations, and went to this conclusion.
That's why I sent a PR, I don't see the point of specifying a default since your code works well without these two lines

**Comment by CharlieEtienne** [comment]
From the docs:

> If no constraint is specified, all types are returned.

Also, is there any good reason to default with 'geocode'?

**Comment by mabdelfattah** [comment]
@CharlieEtienne  Is there a way to implement your overrides locally until the maintainer accepts the pull request?

**Comment by CharlieEtienne** [comment]
> @CharlieEtienne Is there a way to implement your overrides locally until the maintainer accepts the pull request?

@mabdelfattah yes, in the vendor files, see [the changes here](https://github.com/cheesegrits/filament-google-maps/pull/89/files), it's very simple, just a few lines to remove/comment out in two files

**Comment by mabdelfattah** [comment]
> @mabdelfattah yes, in the vendor files, see [the changes here](https://github.com/cheesegrits/filament-google-maps/pull/89/files), it's very simple, just a few lines to remove/comment out in two files

@CharlieEtienne Yeah, but this is almost impossible on production 😊 so probably I need to clone both files to my project and then use them instead of using the packages classes.

**Comment by kengraversen** [comment]
I have successfully used the package "cweagans/composer-patches" to apply patch files for the two mentioned files.
I created the patch files using the git diff function and placed them in a separate directory.
This way I separate the original (and otherwise excellent) package and the focused fix (while we wait for the official fix).
AND - since I am no expert in this field, I didn't even know the composer-patches package, I got some help from ChatGPT ;-)

**Comment by CharlieEtienne** [comment]
@kengraversen 🤔 Why don't you just copy paste the `Map` or `Geocomplete` class and use yours instead of the plugin's one?

```php
// use Cheesegrits\FilamentGoogleMaps\Fields\Map;
// use Cheesegrits\FilamentGoogleMaps\Fields\Geocomplete;
use App\Overrides\Map;
use App\Overrides\Geocomplete;
```

**Comment by kengraversen** [comment]
> @kengraversen 🤔 Why don't you just copy paste the `Map` class and use yours instead of the plugin's one?
> 
> ```
> use App\Overrides\Map;
> ```

Honestly, I wouldn't know how to do that while keeping my code structure clean and controlled.
I see your point - and also that it may be apparently easier - but the way I have done it will ensure that I can update the original package without losing my patches (I believe).

**Comment by tudorcrisan1231** [comment]
> [@CharlieEtienne](https://github.com/CharlieEtienne) Is there a way to implement your overrides locally until the maintainer accepts the pull request?

hello, I created a new Geocomplete class where I extend the original one and then I overwrite the getTypes function


```
<?php

namespace App\Classes;

use Cheesegrits\FilamentGoogleMaps\Fields\Geocomplete as FilamentGeocomplete;

class Geocomplete extends FilamentGeocomplete
{
    public function getTypes(): array
    {
        $types = $this->evaluate($this->types);

        //commented out the following line so that the geocode type is not added to the types array (no limitation on search)
//        if (count($types) === 0) {
//            $types = ['geocode'];
//        }

        return $types;
    }
}
```


and when I want to use the Geocomplete field I just import my class:
`use App\Classes\Geocomplete;`
